crates/ty_python_semantic/src/types.rs

```diff
@@ -6895,6 +6938,46 @@ impl<'db> FunctionType<'db> {
         }
     }
 
+    fn is_subtype_of(self, db: &'db dyn Db, other: Self) -> bool {
+        // A function literal is the subtype of itself, and not of any other function literal.
+        // However, our representation of a function literal includes any specialization that
+        // should be applied to the signature. Different specializations of the same function
+        // literal are only subtypes of each other if they result in the same signature. (Note that
+        // the equality check above will have already handled the case where `self` and `target`
+        // are the same function literal with the same specialization.)
```

Comment:
nit: can you specify which "equality check" this is referring to? Is it the `is_equivalent_to` check in `is_subtype_of`?

Reply:
Oh whoops!  Copy/pasted too much from `Type::is_subtype_of` :hand_over_mouth:  There's no equality check here, so I've removed that part of the comment